p2p/src/lib.rs

```diff
@@ -58,7 +58,52 @@ pub use self::{address::Address, network_ext::NetworkExt};
 /// 70001 - Support bloom filter messages `filterload`, `filterclear` `filteradd`, `merkleblock` and FILTERED_BLOCK inventory type
 /// 60002 - Support `mempool` message
 /// 60001 - Support `pong` message and nonce in `ping` message
-pub const PROTOCOL_VERSION: u32 = 70001;
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ProtocolVersion(u32);
+
+impl ProtocolVersion {
+    /// Support receiving `wtxidrelay` message between `version` and `verack` message
+    pub const WTXID_RELAY_VERSION: ProtocolVersion = ProtocolVersion(70016);
+    /// Support receiving invalid compact blocks from a peer without banning them
+    pub const INVALID_CB_NO_BAN_VERSION: ProtocolVersion = ProtocolVersion(70015);
+    /// Support compact block messages `sendcmpct`, `cmpctblock`, `getblocktxn` and `blocktxn`
+    pub const SHORT_IDS_BLOCKS_VERSION: ProtocolVersion = ProtocolVersion(70014);
+    /// Support `feefilter` message
+    pub const FEEFILTER_VERSION: ProtocolVersion = ProtocolVersion(70013);
+    /// Support `sendheaders` message and announce new blocks via headers rather than inv
+    pub const SENDHEADERS_VERSION: ProtocolVersion = ProtocolVersion(70012);
+    /// Support `pong` message and nonce in `ping` message
+    pub const BIP0031_VERSION: ProtocolVersion = ProtocolVersion(60000);
+    /// All connections will be terminated below this version.
+    pub const MIN_PEER_PROTO_VERSION: ProtocolVersion = ProtocolVersion(31800);
+}
+
+impl ProtocolVersion {
+    /// Construct a protocol version that is not well-known.
+    pub fn from_nonstandard(version: u32) -> Self {
+        Self(version)
+    }
+}
+
+impl Default for ProtocolVersion {
+    fn default() -> Self {
+        Self::WTXID_RELAY_VERSION
+    }
```

Comment:
Its not obvious to me why this is the default. Perhaps its cleaner to not have this function and explicitly use the const?